### PR TITLE
JVMChaos: support inject fault into MySQL client

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve?

close issue #2977 

### Related changes

- [ ] Need to update `chaos-mesh/website`
- [ ] Need to update `Dashboard UI`
- Need to **cheery-pick to release branches**
  - [x] release-2.1
  - [ ] release-2.0

### Checklist

CHANGELOG

- [ ] I have updated the `CHANGELOG.md`
- [ ] I have labeled this PR with "no-need-update-changelog"

Tests

- [ ] Unit test
- [x] E2E test
- [ ] No code
- [ ] Manual test (add steps below)

Side effects

- [ ] Breaking backward compatibility

### Release note 

```text
Please add a release note.

You can safely ignore this section if you don't think this PR needs a release note.
```

### DCO

If you find the DCO check fails, please run commands like below (Depends on the actual situations. For example, if the failed commit isn't the most recent) to fix it:

```shell
git commit --amend --signoff
git push --force
```

### CODE REVIEW

File: api/v1alpha1/jvmchaos_types.go [hunk 1]
```diff
@@ -122,6 +129,29 @@ type JVMStressCfgSpec struct {
 	MemoryType string `json:"memType,omitempty"`
 }
 
+// JVMMySQLSpec is the specification of MySQL fault injection in JVM
+// only when SQL match the Database, Table and SQLType, JVMChaos mesh will inject fault
+// for examle:
+//   SQL is "select * from test.t1",
+//   only when ((Database == "test" || Database == "") && (Table == "t1" || Table == "") && (SQLType == "select" || SQLType == "")) is true, JVMChaos will inject fault
+type JVMMySQLSpec struct {
+	// the version of mysql-connector-java, only support 5.X.X(set to 5) and 8.X.X(set to 8) now
```
Comment (STRRL) [inline api/v1alpha1/jvmchaos_types.go:138]:
```suggestion
	// the version of mysql-connector-java, only support 5.X.X(set to "5") and 8.X.X(set to "8") now
```

using a double quote to emphasize here requires a `string`

File: api/v1alpha1/jvmchaos_webhook.go [hunk 2]
```diff
@@ -79,6 +79,13 @@ func (in *JVMChaosSpec) Validate(root interface{}, path *field.Path) field.Error
 		if len(in.RuleData) == 0 {
 			allErrs = append(allErrs, field.Invalid(path, in, "rule data not provide"))
 		}
+	case JVMMySQLAction:
+		if len(in. MySQLConnectorVersion) == 0 {
+			allErrs = append(allErrs, field.Invalid(path, in, "MySQL connector version not provided"))
+		}
```
Comment (STRRL) [inline api/v1alpha1/jvmchaos_webhook.go:85]: also check the version could only be `"5"` or `"8"`?

**Review by STRRL — COMMENTED** [review]
rest LGTM! ❤️

File: api/v1alpha1/jvmchaos_webhook.go [hunk 3]
```diff
@@ -79,6 +79,13 @@ func (in *JVMChaosSpec) Validate(root interface{}, path *field.Path) field.Error
 		if len(in.RuleData) == 0 {
 			allErrs = append(allErrs, field.Invalid(path, in, "rule data not provide"))
 		}
+	case JVMMySQLAction:
+		if len(in. MySQLConnectorVersion) == 0 {
```
Comment (STRRL) [inline api/v1alpha1/jvmchaos_webhook.go:83]:
```suggestion
		if len(in.MySQLConnectorVersion) == 0 {
```

**Comment by WangXiangUSTC** [comment]
There is something wrong with my git tree and it's difficult to resolve, I will open another pr

**Pull request closed without merging**